src/main/java/org/folio/entlinks/domain/repository/AuthoritySourceFileCodeRepository.java

```diff
@@ -3,8 +3,12 @@
 import java.util.Optional;
 import org.folio.entlinks.domain.entity.AuthoritySourceFileCode;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface AuthoritySourceFileCodeRepository extends JpaRepository<AuthoritySourceFileCode, Integer> {
 
-  Optional<AuthoritySourceFileCode> findFirstByCodeStartsWith(String code);
+  @Query(value = "SELECT * FROM authority_source_file_code sfc where :pattern like (sfc.code || '%') LIMIT 1",
+      nativeQuery = true)
```

Comment:
This query has to be improved because there could be such cases:
![image](https://github.com/folio-org/mod-entities-links/assets/5904625/05e43c80-bb3e-4df4-b616-11ba67ae9c12)
The query found an `n` code but had to find `ns` code.
So I suggest ordering codes by length:
![image](https://github.com/folio-org/mod-entities-links/assets/5904625/42ef0c4b-7cfb-400e-aa0a-f249d76ffeed)

Reply:
Also I'm curious why we have such ids for the codes table

Reply:
Applied suggested ordering

Reply:
IDs of the authority_source_file_code table are also fixed:

![image](https://github.com/folio-org/mod-entities-links/assets/133661057/5f533150-ad80-40b1-9112-5dfed1eb259c)